src/cpp/thread_manager/thread_manager.cc

```diff
@@ -34,15 +34,28 @@ ThreadManager::WorkerThread::WorkerThread(ThreadManager* thd_mgr)
   thd_ = grpc_core::Thread(
       "grpcpp_sync_server",
       [](void* th) { static_cast<ThreadManager::WorkerThread*>(th)->Run(); },
-      this);
-  thd_.Start();
+      this,
+      &created_
+      );
+  if (!created_) {
+    gpr_log(GPR_ERROR,
+            "Could not create grpc_sync_server worker-thread"
+            );
+  } else {
+    thd_.Start();
+  }
 }
 
 void ThreadManager::WorkerThread::Run() {
   thd_mgr_->MainWorkLoop();
   thd_mgr_->MarkAsCompleted(this);
 }
 
+
+bool ThreadManager::WorkerThread::isCreated() {
+  return created_;
+}
```

Comment:
I think we can move this defn inline (and i should probably do so for the other members of `ThreadManager::WorkerThread` in a separate PR). I've also suggested a new name in my comment on the header file.

Reply:
No, inline in a .cc won't work in general (though I suppose it's ok in this case since the use is in the same translation unit). I just meant moving it into the .h file so it looks like

```cpp
   bool created() const { return created_; }
```

Reply:
Renamed isCreated, inlined it and labeled it const, as suggested. :)

Reply:
Just read your comment... expect a new commit in a few minutes... :)

Reply:
Thanks for putting up with my comments. I'd like to get this in.

Reply:
You're welcome - I also want to do this the right way... :)